commit 7786ce823b1c9a3de01a63857e3451890cb4e81c
Author: Jie Zhang <jie.zhang@analog.com>
Date:   Thu Mar 5 18:50:26 2009 +0800

Blackfin arch: fix bug - gdb signull case make trunk kernel panic frequently

Use copy_to_user_page and copy_from_user_page instead of
memcpy. copy_to_user_page does cache flush when necessary.

Signed-off-by: Jie Zhang <jie.zhang@analog.com>
Signed-off-by: Bryan Wu <cooloney@kernel.org>

diff --git a/arch/blackfin/kernel/ptrace.c b/arch/blackfin/kernel/ptrace.c
index 594e325..d76618d 100644
--- a/arch/blackfin/kernel/ptrace.c
+++ b/arch/blackfin/kernel/ptrace.c
@@ -240,7 +241,7 @@
 
 } else if (addr >= FIXED_CODE_START
     && addr + sizeof(tmp) <= FIXED_CODE_END) {
-	memcpy(&tmp, (const void *)(addr), sizeof(tmp));
+	copy_from_user_page(0, 0, 0, &tmp, (const void *)addr, sizeof(tmp));
 	copied = sizeof(tmp);
 
 } else
@@ -320,7 +321,7 @@
 
 } else if (addr >= FIXED_CODE_START
     && addr + sizeof(data) <= FIXED_CODE_END) {
-	memcpy((void *)(addr), &data, sizeof(data));
+	copy_to_user_page(0, 0, 0, (void *)addr, &data, sizeof(data));
 	copied = sizeof(data);
 
 } else
